src/Stratis.Bitcoin.Features.MemoryPool/MempoolBehavior.cs

```diff
@@ -487,6 +487,12 @@ public void RelayTransaction(uint256 hash)
             IEnumerable<MempoolBehavior> behaviours = peers.Select(s => s.Behavior<MempoolBehavior>());
             foreach (MempoolBehavior mempoolBehavior in behaviours)
             {
+                if (mempoolBehavior?.AttachedPeer == null)
+                {
+                    this.logger.LogTrace("Peer is null, skipped.");
+                    continue;
```

Comment:
I don't think `continue;` is correct if we have no attached peer no point to continue.

Reply:
you can still have race conditions later down the line,
I suggest taking a local var of the peer

```
var peer = mempoolBehavior?.AttachedPeer

if(peer == null)
  continue
```
form here us `peer`

Reply:
Fixed 👍